pika/adapters/select_connection.py

```diff
@@ -264,21 +264,38 @@ def get_next_deadline(self):
         timeout = min((timeout, SelectPoller.POLL_TIMEOUT))
         return timeout * SelectPoller.POLL_TIMEOUT_MULT
 
+    @staticmethod
+    def is_ready_to_fire(to_run):
+        """For patching by unittest"""
+        return True
+
     def process_timeouts(self):
         """Process the self._timeouts event stack"""
 
         now = time.time()
-        to_run = [timer for timer in self._timeouts.values()
-                  if timer['deadline'] <= now]
-
         # Run the timeouts in order of deadlines. Although this shouldn't
         # be strictly necessary it preserves old behaviour when timeouts
         # were only run periodically.
-        for t in sorted(to_run, key=itemgetter('deadline')):
```

Comment:
Something like: `W0201: Attribute 'timer_stack' defined outside __init__` should be disabled globally in this script. With `unittest`, you typically don't declare your own `__init__` in the test case classes. Since this is normal behavior for unittest class implementations, disable this warning globally in the specific script. I tend to put the disable-warning pragmas between the module's docstring and the first import for visibility:

Using this script as an example:
```
# -*- coding: utf8 -*-
"""
Tests for SelectConnection IOLoops

"""
# pylint: disable=W0201

import logging

...
```

Please note that it is generally bad practice to disable pylint warnings globally like this, because that hides potential bugs. In most cases, you need to fix the code; in very special cases, where you really need to disable a pylint warning for a good reason, you would disable it on a case-by-case basis; you can put the pragma at the end of a statement to disable the warning on a specific line of code;  you can also use disable/enable pragmas around a block of code, if necessary.